subprojects/core-api/src/main/java/org/gradle/api/file/ContentFilterable.java

```diff
@@ -85,9 +87,36 @@
      * Groovy's {@link groovy.text.SimpleTemplateEngine}. This means you can use simple property references, such as
      * <code>$property</code> or <code>${property}</code> in the file. You can also include arbitrary Groovy code in the
      * file, such as <code>${version ?: 'unknown'}</code> or <code>${classpath*.name.join(' ')}</code>
+     * <p>
+     * Note that all escape sequences ({@code \n}, {@code \t}, {@code \\}, etc) are converted to the symbols
+     * they represent, so, for example, {@code \n} becomes newline. If this is undesirable then {@link #expand(Map, Action)}
+     * should be used to disable this behavior:
+     * <pre>
+     *  expand(one: '1', two: 2) {
+     *      escapeBackslash = true
+     *  }
+     * </pre>
      *
      * @param properties to implement line based filtering
      * @return this
      */
     ContentFilterable expand(Map<String, ?> properties);
+
+    /**
+     * <p>Expands property references in each file as it is copied. More specifically, each file is transformed using
+     * Groovy's {@link groovy.text.SimpleTemplateEngine}. This means you can use simple property references, such as
+     * <code>$property</code> or <code>${property}</code> in the file. You can also include arbitrary Groovy code in the
+     * file, such as <code>${version ?: 'unknown'}</code> or <code>${classpath*.name.join(' ')}</code>
+     * <p>
+     * Note that by default all escape sequences ({@code \n}, {@code \t}, {@code \\}, etc) are converted to the symbols
+     * they represent, so, for example, {@code \n} becomes newline. This behavior is controlled by
+     * {@link ExpandDetails#getEscapeBackslash()} property.
+     *
+     * @param properties to implement line based filtering
+     * @param action action to perform additional configuration of the underlying template engine
+     * @return this
+     * @since 7.2
+     */
+    @Incubating
+    ContentFilterable expand(Map<String, ?> properties, Action<? super ExpandDetails> action);
```

Comment:
Can you reword this a bit so that the first paragraph here is not identical to `#expand(Map)`?  That way they won't show up as identical in the table here: https://builds.gradle.org/repository/download/Gradle_Master_Check_BuildDistributions/44719298:id/distributions/gradle-7.2-docs.zip!/gradle-7.2-20210709145511%2B0000/docs/dsl/org.gradle.api.tasks.Copy.html#N20B99

Reply:
I mentioned `action` in the first paragraph. BTW, that is an interesting link, I wasn't aware that it is possible to look into the generated docs. Thanks!